contracts/vault/src/common.rs

```diff
@@ -85,3 +88,9 @@ pub fn check_not_paused(config: &Config) -> Result<()> {
     }
     Ok(())
 }
+
+pub fn get_and_increment_queue_id(store: &mut dyn Storage) -> StdResult<QueueId> {
+    let counter = state::QUEUE_COUNTER.load(store)?;
```

Comment:
Use 
```rust
let counter = state::QUEUE_COUNTER.may_load(store)?.unwrap_or_default();
```
 instead which returns an `Option`. Otherwise, `load` will just straight up throw an error the first time this is used because there was no initial value written before. This unfortunately is just one of the shortcomings of their API design and literally everyone (including me) falls for this when they are first starting out...